### PR TITLE
build: Align on tools from `.versions` also for the build / test workflow

### PR DESCRIPTION
**Pull request by sschuberth**

### CODE REVIEW

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -96,6 +92,8 @@ jobs:
     needs: build-web-app-reporter
     runs-on: ubuntu-22.04
     steps:
+    - name: Set tool version environment variables
```
Comment (fviernau) [inline .github/workflows/build-and-test.yml:95]: Does this commit actually align (change) any tool version?

Comment (sschuberth) [reply .github/workflows/build-and-test.yml:95]: Yes, the ScanCode version.

Comment (sschuberth) [reply .github/workflows/build-and-test.yml:95]: But weird that `.versions` is not found, as this step `needs: build-web-app-reporter`, so a checkout should be available...

Comment (sschuberth) [reply .github/workflows/build-and-test.yml:100]: Nevermind, different jobs do not share the workspace.

**Comment by sschuberth** [comment]
Blocked by https://github.com/oss-review-toolkit/ort/pull/7883.

**Review by fviernau — APPROVED** [review]

**Pull request merged**